### ISSUE TITLE
[Accessibility]: Cannot navigate to line numbers using keyboard navigation

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36528]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34448

### Problem description

Right now, the blob view is in a [table](https://github.com/sourcegraph/sourcegraph/issues/36521) inside a group. It doesn't seem possible to navigate to the line numbers at all using keyboard navigation (not including screen reader controls).

This means that it is not possible to "click" on a line number, and generate a canonical link, similar to https://github.com/sourcegraph/sourcegraph/issues/36526

### Expected behavior

It should be possible to navigate to the line numbers, so that one can select a single line or a range of lines.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
